src/compat.rs

```diff
@@ -0,0 +1,224 @@
+// Things of the past, mostly bad decisions, which cannot erased, stays in this
+// haunted module.
```

Comment:
😂 this is great, you just made my day. Maybe we can display warnings the first time someone uses these after the update telling them to migrate because they will be removed in a future version?

Reply:
I don't think it will be removed, atleast anytime soon. Maybe after a few major releases, when there are more plugins using the newer apis, might revisit this.